app/code/Magento/Catalog/Block/Adminhtml/Category/Tab/Product.php

```diff
@@ -159,6 +165,30 @@ protected function _prepareColumns()
         );
         $this->addColumn('name', ['header' => __('Name'), 'index' => 'name']);
         $this->addColumn('sku', ['header' => __('SKU'), 'index' => 'sku']);
+
+
+        $this->addColumn(
+            'visibility',
+            [
+                'header' => __('Visibility'),
+                'index' => 'visibility',
+                'type' => 'options',
+                'options' => Visibility::getOptionArray(),
+                'header_css_class' => 'col-visibility',
+                'column_css_class' => 'col-visibility'
+            ]
+        );
+
+        $this->addColumn(
+            'status',
+            [
+                'header' => __('Status'),
+                'index' => 'status',
+                'type' => 'options',
+                'options' => Status::getOptionArray()
```

Comment:
Same as Visibility

Reply:
I guess it shouldn't be static, it's a piece of inherited code from Magento 1, and it should be refactored at some point: https://github.com/magento/magento2/blob/0849373d7636cfa17b70b8188bfc36d13ae749c1/app/code/Magento/Catalog/Model/Product/Visibility.php

About using the object manager, it's true that most of time shouldn't be used directly, but constructor params BC may be one of the few cases where it could make sense, since when the major version is increased and no BC is required, all constructors are refactored to remove this direct object manager usage. From my point of view, where using object manager is absolutely forbidden is in the main logic of any module, if object manager is strictly required to make the module work.